effekt/shared/src/main/scala/effekt/Typer.scala

```diff
@@ -1239,8 +1244,16 @@ object Typer extends Phase[NameResolved, Typechecked] {
 
     var effs: ConcreteEffects = Pure
 
+    (vpnames.map(Some.apply).zipAll(vargs, None, source.ValueArg.Unnamed(source.UnitLit(source.Span.missing)))) foreach {
+      case (Some(expName), source.ValueArg(Some(gotName), _, _)) if expName != gotName =>
+        Context.error(s"Unexpected name in named argument: got ${gotName}, but expected ${expName}.")
```

Comment:
Nitpick: could we also add some `if gotName in vargs.map { ...name }` condition to say something reminding the user that there _is_ an arg with this name, but the _position_ is wrong.